app/services/base_api_client.rb

```diff
@@ -138,9 +138,12 @@ def parse_response_to_struct(response, struct)
 
   def adapt_to_struct(struct, obj)
     out = {}
-
     struct.props.keys.each do |key|
-      out[key] = obj.fetch(key, nil)
+      if obj.fetch(key, nil).is_a?(Hash)
+        out[key] = adapt_to_struct(struct.props[key][:type], obj.fetch(key, nil))
```

Comment:
oh my, much recursion.

Reply:
@jlbyrne What is the `:type` symbol?

Reply:
If you take a look at the struct, each key has its type listed:
```
{:uphold=>{:immutable=>true, :type=>Rewards::Types::AllowList, :type_object=>#<T::Types::Simple:0x0000aaab0fe2a850 @raw_type=Rewards::Types::AllowList>, :accessor_key=>:@uphold, :sensitivity=>nil, :pii=>nil, :extra=>nil, :setter_proc=>#<Proc:0x0000aaab0fe299a0 /usr/local/bundle/gems/sorbet-runtime-0.5.9538/lib/types/props/private/setter_factory.rb:61>, :serialized_form=>"uphold", :fully_optional=>false, :need_nil_read_check=>nil}
```
I'm using that to set the value of struct in the recursive call.  Unfortunately, there's no good way to extract whether or not the key's type is a primitive like a string, or something custom.

The recursion is really unfortunate, but I don't see another way of handling a utility function for any kind of nested JSON. 😢

Reply:
We're not going to be able to hard code the `:type` symbol in that method as it is a global handler.  It will look for a type value in all subsequent hash responses.

Reply:
Let me think about it.  The response object for params is rough.